### ISSUE TITLE
How to check a checkbox?

### ISSUE THREAD

**Issue by camdez** [igrishaev/etaoin#366]
What is the recommended way to check a checkbox (`<input type="checkbox">`)? I've tried every variation of `etaoin.api/fill` values that I could think of, to no avail, and been through the API docs a few times looking for a more specific function.

The best I've managed is using the spacebar to _toggle_ the value:

```clj
(defn- toggle-checkbox [driver q]
  (etaoin.api/fill driver q etaoin.keys/space))
```

...but that approach tends to assume the initial state of the checkbox, which is not ideal for testing.

Thanks!

**Comment by Uunnamed** [comment]
Hi, @camdez 

Have you tried just `click`ing?

`(click driver q)`

**Comment by camdez** [comment]
Hi @Uunnamed,

That's actually the first thing I tried. I don't know exactly why, but I get an exception...

```
"element click intercepted: Element <input type=\"checkbox\" name=\"accept-terms?\" class=\"custom-control-input\"> is not clickable at point (473, 678). Other element would receive the click: <span class=\"custom-control-label\">...</span>\n  (Session info: chrome=91.0.4472.114)\n  (Driver info: chromedriver=91.0.4472.101 (af52a90bf87030dd1523486a1cd3ae25c5d76c9b-refs/branch-heads/4472@{#1462}),platform=Mac OS X 11.4.0 x86_64)"
```

For some reason the containing label seems to steal the click.

I spent a little time trying to work back to a minimal example and then work forward again...I do see that a bare checkbox *is* `click`able, as is an unstyled checkbox wrapped in a `<label>`. Not sure why my particular styling seems to break things.

Regardless, it does still make sense to me to be able to (idempotently) *set* the value rather than toggling it with a click. A user will be looking to set a checkbox to a certain value most of the time, and it feels like we should be able to replicate that in a test.

**Comment by camdez** [comment]
Happy to submit a PR on this too, just want to see if there's agreement on the approach / if there is a pre-existing solution I haven't thought of.

**Comment by whamilton42** [comment]
It looks like the label is in the way. Most sites nowadays let you click the label in order to click the checkbox - have you tried that?

**Comment by camdez** [comment]
@whamilton42 Yep! I think it would generally work, but my label contains a link, and clicking the label via etaoin seems to activate the link instead of clicking the unlinked portion.

Lest that all sound crazy, this is actually all I'm dealing with:

<img width="303" alt="Screen Shot 2021-07-08 at 10 21 26 AM" src="https://user-images.githubusercontent.com/141447/125817407-0a124443-abf2-48b5-82ae-f6891e00b01b.png">
